foreman/data_refinery_foreman/foreman/main.py

```diff
@@ -119,6 +137,123 @@ def get_active_volumes() -> Set[str]:
+    9/10 samples in an experiment are processed then they can't
+    actually be used until that last sample is processed.
+    """
+    # The strategy for doing so is to build a mapping between every
+    # salmon job in `jobs` to a priority. This priority will be what
+    # percentage of the samples in this experiment have been
+    # processed. Once we have that mapping we can sort those jobs by
+    # that priority and move them to the front of the list.
+    prioritized_jobs = []
+    for job in jobs:
+        # Salmon jobs are specifc to one sample.
+        sample = job.get_samples().pop()
+
+        # Skip jobs that aren't for Salmon. Handle both ProcessorJobs and DownloaderJobs.
+        if type(job) is ProcessorJob and job.pipeline_applied != ProcessorPipeline.SALMON.value:
+            continue
+        elif type(job) is DownloaderJob:
+            is_salmon_sample = False
+            for original_file in sample.original_files.all():
+                if original_file.filename[-5:].upper() == "FASTQ" \
+                or original_file.filename[-8:].upper() == "FASTQ.GZ" \
+                or original_file.filename[-2:].upper() == "FQ" \
+                or original_file.filename[-3:].upper() == "SRA" \
+                or original_file.filename[-5:].upper() == "FQ.GZ":
+                    is_salmon_sample = True
+
+            if not is_salmon_sample:
+                continue
+
+        # Get a set of unique samples that share at least one
+        # experiment with the sample this job is for.
+        related_samples = set()
+        for experiment in sample.experiments.all():
+            for related_sample in experiment.samples.all():
+                related_samples.add(related_sample)
+
+        # We cannot simply filter on is_processed because that field
+        # doesn't get set until every sample in an experiment is processed.
+        # Instead we are looking for one successful processor job.
+        processed_samples = 0
+        for related_sample in related_samples:
+            sample_is_processed = False
+            for processor_job in related_sample.original_files.first().processor_jobs.all():
+                if processor_job.success == True:
+                    sample_is_processed = True
+
+            if sample_is_processed:
+                processed_samples += 1
+
+        experiment_completion_percent = processed_samples / len(related_samples)
+        prioritized_jobs.append({"job": job, "priority": experiment_completion_percent})
+
+    sorted_job_mappings = sorted(prioritized_jobs, reverse=True, key=lambda k: k["priority"])
+    sorted_jobs = [job_mapping["job"] for job_mapping in sorted_job_mappings]
+
+    # Remove all the jobs we're moving to the front of the list
+    for job in sorted_jobs:
+        jobs.remove(job)
+
+    return sorted_jobs + jobs
+
+
+def prioritize_zebrafish_jobs(jobs: List) -> List:
+    zebrafish_jobs = []
```

Comment:
docstring to explain what this does - is it sorting a list? pruning a list? etc. Same for function below